src/metal/window-proxy.ts

```diff
@@ -68,37 +88,32 @@ function windowSetDimensionsMeta() {
   W.meta.width = W.getWidth();
 }
 
+// Memoize window meta scroll position
 function windowSetScrollMeta() {
   W.meta.scrollLeft = W.getScrollLeft();
   W.meta.scrollTop = W.getScrollTop();
 }
 
-// Only invalidate window dimensions on resize
-function resizeThrottle() {
-  window.clearTimeout(resizeTimeout);
-
-  resizeTimeout = window.setTimeout(() => {
-    windowSetDimensionsMeta();
-  }, throttleDelay);
-
-  W.onWindowIsDirtyListeners.forEach((obj) => {
-    let { fn, scope } = obj;
-    fn.call(scope);
-  });
-}
-
-// Only invalidate window scroll on scroll
-function scrollThrottle() {
-  window.clearTimeout(scrollTimeout);
-
-  scrollTimeout = window.setTimeout(() => {
-    windowSetScrollMeta();
-  }, throttleDelay);
+// Only invalidate window isDirty on scroll and resize
+function eventThrottle(eventType?: string, root?: Element) {
```

Comment:
Does consolidating into one throttle function buy us anything if we are still switching on each case?

Reply:
Nothing major. One less function declaration in a succinct easier to read package.

Reply:
I would strongly recommend using 2 discrete methods.

Rationale (for 2 functions)

* more concise (less bytes)
* less prone to error (if an invalid event type is used, we get a useful error)
* remove unneeded wrapper function
* we should defer machinery for extensibility until not having that machinery is causes grief.

Reply:
also the second any of `root` appears to be never used, we can likely drop.

Reply:
If a throttle utility is to be used, we should do so to avoid extra bytes. I believe something like the following could be used:

```js
function throttle(cb) {
  let cookie;
  return () => {
    clearTimeout(cookie);
    cookie = setTimeout(cb, throttleDelay)
    // anything else, maybe the dirty check thing.
  }
}

addEventListener('resize', throttle(windowSetDimensionsMeta))
addEventListener('scroll', throttle(windowSetScrollMeta));
```